singer_sdk/connectors/sql.py

```diff
@@ -81,34 +91,31 @@ def create_sqlalchemy_connection(self) -> sqlalchemy.engine.Connection:
         Returns:
             A newly created SQLAlchemy engine object.
         """
-        return (
-            self.create_sqlalchemy_engine()
-            .connect()
-            .execution_options(stream_results=True)
-        )
+        return self._engine.connect().execution_options(stream_results=True)
 
-    def create_sqlalchemy_engine(self) -> sqlalchemy.engine.Engine:
-        """Return a new SQLAlchemy engine using the provided config.
+    def create_sqlalchemy_engine(self) -> Engine:
+        """(DEPRECATED) Return a new SQLAlchemy engine using the provided config.
 
         Developers can generally override just one of the following:
         `sqlalchemy_engine`, sqlalchemy_url`.
 
         Returns:
             A newly created SQLAlchemy engine object.
         """
-        return sqlalchemy.create_engine(self.sqlalchemy_url, echo=False)
+        return self._engine
```

Comment:
```suggestion
        warnings.warn(
            "`create_sqlalchemy_engine` is deprecated. Override `sqlalchemy_engine` or sqlalchemy_url` instead.",
            DeprecationWarning
        )
        return self._engine
```

Reply:
Yeah that's much better than a warning in the docstring haha. Thanks!